src/genworld_gui.cpp

```diff
@@ -968,7 +968,7 @@ struct CreateScenarioWindow : public Window
 		this->SetWidgetDisabledState(WID_CS_START_DATE_DOWN,       _settings_newgame.game_creation.starting_year <= MIN_YEAR);
 		this->SetWidgetDisabledState(WID_CS_START_DATE_UP,         _settings_newgame.game_creation.starting_year >= MAX_YEAR);
 		this->SetWidgetDisabledState(WID_CS_FLAT_LAND_HEIGHT_DOWN, _settings_newgame.game_creation.se_flat_world_height <= 0);
-		this->SetWidgetDisabledState(WID_CS_FLAT_LAND_HEIGHT_UP,   _settings_newgame.game_creation.se_flat_world_height >= MAX_TILE_HEIGHT);
+		this->SetWidgetDisabledState(WID_CS_FLAT_LAND_HEIGHT_UP,   _settings_newgame.game_creation.se_flat_world_height >= (_settings_newgame.construction.map_height_limit != 0 ? _settings_newgame.construction.map_height_limit : MAX_MAP_HEIGHT_LIMIT - 1));
```

Comment:
MAX_MAP_HEIGHT_LIMIT is still a valid value, so no "- 1".

(same thing repeats a few times below)